tests/quality_pragmaitc/test_refinement.py

```diff
@@ -24,63 +32,430 @@ def test_shouldBeApplicable():
+                           15, Comparison.LESS_OR_EQUAL_TO)
+
+    task1.addApplicableContext(current)
+    task1.setProvidedQuality(current, CommonMetrics.SECONDS, 16)
+
+    task2.addApplicableContext(current)
+    task2.setProvidedQuality(current, CommonMetrics.SECONDS, 11)
+
+    goal.addDependency(task1)
+    goal.addDependency(task2)
+
+    goal.setIdentifier("Root")
+    goal.addApplicableContext(current)
+
+    interp = Interpretation()
+    interp.addQualityConstraint(qc)
+
+    plan = goal.isAchievable(fullContext, interp)
 
-    def test_shouldBeNotApplicable():
-        #goal = Goal()
-        task = Task()
-        delegation = Delegation()
+    assert plan is None
 
-        context = Context("C1")
 
-        task.addApplicableContext(context)
-        #goal.addApplicableContext(context)
-        delegation.addApplicableContext(context)
+def test_aGoalOrDecomposedWithTwoTasksMayBeAchievable():
+    goal = Goal(Decomposition.OR)
+    assert goal.decomposition
 
-        wrongContext = Context("C2")
-        fullContext = []
-        fullContext.add(wrongContext)
+    task1 = Task()
+    task2 = Task()
 
-        #assert False == goal.isApplicable(fullContext)
-        assert False == task.isApplicable(fullContext)
-        assert False == delegation.isApplicable(fullContext)
+    current = Context("C1")
+    fullContext = []
+    fullContext.append(current)
+
+    qc = QualityConstraint(current, CommonMetrics.SECONDS,
+                           15, Comparison.LESS_OR_EQUAL_TO)
+
+    task1.addApplicableContext(current)
+    task1.setProvidedQuality(current, CommonMetrics.SECONDS, 13)
+
+    task2.addApplicableContext(current)
+    task2.setProvidedQuality(current, CommonMetrics.SECONDS, 11)
+
+    goal.addDependency(task1)
+    goal.addDependency(task2)
+
+    goal.setIdentifier("Root")
+    goal.addApplicableContext(current)
+
+    interp = Interpretation()
+    interp.addQualityConstraint(qc)
+
+    plan = goal.isAchievable(fullContext, interp)
+    assert len(plan.getTasks()) == 1
+
+
+def test_aGoalOrDecomposedWithTwoTasksMayBeAchievableAtOnlyOneBranch():
+    goal = Goal(Decomposition.OR)
+    assert goal.decomposition
+
+    task1 = Task()
+    task2 = Task()
+
+    current = Context("C1")
+    fullContext = []
+    fullContext.append(current)
+
+    qc = QualityConstraint(current, CommonMetrics.SECONDS,
+                           15, Comparison.LESS_OR_EQUAL_TO)
+
+    task1.addApplicableContext(current)
+    task1.setProvidedQuality(current, CommonMetrics.SECONDS, 16)
+
+    task2.addApplicableContext(current)
+    task2.setProvidedQuality(current, CommonMetrics.SECONDS, 11)
+
+    goal.addDependency(task1)
+    goal.addDependency(task2)
+
+    goal.setIdentifier("Root")
+    goal.addApplicableContext(current)
+
+    interp = Interpretation()
+    interp.addQualityConstraint(qc)
+
+    plan = goal.isAchievable(fullContext, interp)
+    assert plan.getTasks().contains(task2)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=4861978)